nuklear.h

```diff
@@ -13922,8 +13922,12 @@ nk_input_key(struct nk_context *ctx, enum nk_keys key, int down)
     NK_ASSERT(ctx);
     if (!ctx) return;
     in = &ctx->input;
+#ifdef NK_KEYSTATE_BASED_INPUT
```

Comment:
Please write a short comment answering the questions "Why is it here? And why does it look like this and not otherwise?".

Reply:
I explain in detail why it's there in the commit message and I think the name is pretty explanatory.  

If you're talking about an actual code comment right there explaining what it does, I think that's overkill and doesn't match the style of the code in nuklear.  There are several examples (ie line 7152, 8691, 8716 etc.) with #ifdef's in the code and no comments, and that's not even counting when entire functions are #ifdefed without comments.

I admit that I should have added a descriptive comment on line 109 where all the configuration macros are documented.  I should have thought to look for that.  I'll do that and resubmit, though it'd be easy enough for anyone with commit access to just add that since I selected the "maintainer can make changes" box.

Reply:
>a descriptive comment on line 109 where all the configuration macros are documented.

This is exactly what I meant, thanks. And sorry for not being clear from the beginning.